python/res/job_queue/job_queue_manager.py

```diff
@@ -155,6 +155,15 @@ def execute_queue(self):
                 started_job_threads.append(job.run(job_queue.driver))
                 job = job_queue.fetch_next_waiting()
             time.sleep(1)
+            if job_queue.stopped_by_user:
+                for job in job_queue.job_list:
+                    job.stop(job_queue.driver)
 
         for thread in started_job_threads:
             thread.join()
+
+        #Cleanup all not-started jobs
+        for job in job_queue.job_list:
+            if job.status == JobStatusType.JOB_QUEUE_WAITING:
+                job._set_status(JobStatusType.JOB_QUEUE_IS_KILLED)
```

Comment:
This looks a bit like a hack. I would guess that `job.stop` handles the status change for a job that is running. I don't see why the jobs that are waiting should be treated differently.
If this is really necessary, the comment should be more exhaustive

Reply:
> This looks a bit like a hack. I would guess that `job.stop` handles the status change for a job that is running. I don't see why the jobs that are waiting should be treated differently.
> If this is really necessary, the comment should be more exhaustive

This's to get the correct status on non started jobs (after being stopped by the user. Agree, will include a comment there.